go/vumitools/tasks.py

```diff
@@ -0,0 +1,53 @@
+import json
+from celery.task import task
+
+from go.base.command_utils import get_user_by_account_key
+from go.base.utils import vumi_api, vumi_api_for_user
+from go.config import get_conversation_definition
+from go.vumitools.metrics import get_conversation_metric_prefix
+
+
+@task(ignore_result=True)
+def send_recent_conversation_metrics():
+    conversation_details = get_and_reset_recent_conversations()
+
+    for conv_details in conversation_details:
+        details = json.loads(conv_details)
+        user_api = get_user_api(details["account_key"])
+        conv = get_conversation(details["account_key"], details["conv_key"])
+        publish_conversation_metrics(user_api, conv)
+
+
+def get_and_reset_recent_conversations():
+    redis = vumi_api().redis.sub_manager("conversation.metrics.middleware")
```

Comment:
This is where you're getting cleanup errors. You're creating a new `VumiApi`, which has its own riak manager, which you're not cleaning up. You can call `api.close()` to cleanup the riak and redis managers. (You will need to do this everywhere you call `vumi_api()` and `vumi_api_for_user()`